addons/pos_restaurant/static/src/js/floors.js

```diff
@@ -100,276 +153,208 @@ const PosRestaurantPosGlobalState = (PosGlobalState) => class PosRestaurantPosGl
-                } else {
-                    self.remove_from_server_and_set_sync_state(ids_to_remove);
-                }
-            }).catch(function(reason){
-                self.set_synch('error');
-            }).finally(function(){
-                return self.clean_table_transfer(table);
+        const timeout = TIMEOUT * removedOrdersIds.length;
+        this.set_synch('connecting', removedOrdersIds.length);
+        try {
+            const removeOrdersResponseData = await this.env.services.rpc({
+                model: 'pos.order',
+                method: 'remove_from_ui',
+                args: [removedOrdersIds],
+            }, {
+                timeout: timeout,
+                shadow: true,
             });
-        } else {
-            if (ids_to_remove.length) {
-                self.remove_from_server_and_set_sync_state(ids_to_remove);
+            this.set_synch('connected');
+            this._postRemoveFromServer(removedOrdersIds, removeOrdersResponseData);
+        } catch (reason) {
+            let error = reason.message;
+            if (error.code === 200) {
+                // Business Logic Error, not a connection problem
+                //if warning do not need to display traceback!!
+                if (error.data.exception_type == 'warning') {
+                    delete error.data.debug;
+                }
             }
-            return self.clean_table_transfer(table);
+            // important to throw error here and let the rendering component handle the error
+            console.warn('Failed to remove orders:', removedOrdersIds);
+            throw error;
         }
     }
 
-    update_table_order(server_id, table_orders) {
-        const order = table_orders.find(o => o.name === server_id.pos_reference);
-        if (order) {
-            order.server_id = server_id.id;
-            order.save_to_db();
-        }
-        return order;
+    // to override
+    _postRemoveFromServer(serverIds, data) {
+        this.db.set_ids_removed_from_server(serverIds);
     }
 
     /**
-     * @param {models.Order} order order to set
+     * Replace all the orders of a table by orders fetched from the backend
+     * @param tableId ID of the table
+     * @throws error
      */
-    set_order_on_table(order) {
-        var orders = this.get_order_list();
-        if (orders.length) {
-            order = order ? orders.find((o) => o.uid === order.uid) : null;
-            if (order) {
-                this.set_order(order);
-            } else {
-                // do not mindlessly set the first order in the list.
-                orders = orders.filter(order => !order.finalized);
-                if (orders.length) {
-                    this.set_order(orders[0]);
-                } else {
-                    this.add_new_order();
-                }
+    async _syncTableOrdersFromServer(tableId) {
+        this._removeOrdersFromServer(); // in case we were offline and we deleted orders in the mean time
```

Comment:
This should be awaited, right? E.g. The deleted order while offline belongs to the `tableId`.

Reply:
Oh correct, I didn't think of that specific use case